### ISSUE TITLE
Problem with update local_config_context on an existing device

### ISSUE THREAD

**Issue by nazarko-es** [digitalocean/pynetbox#242]
Hello, i have problem with update local_config_context on an existing device.
My environment
Python version: 3.8
NetBox version: 2.8.4

Steps to Reproduce
Upload device with local_context_data field. For example my field:
{
"bgp_neighbors": [
{
"as": "65000",
"neighbor": "169.254.0.1"
},
{
"as": "65001",
"neighbor": "169.254.0.2"
}
],
"track": [
"10.10.0.251/32",
"10.10.0.252/32"
],
"vlans": []
}
Get local_context_data in python script and append local_context_data['vlans']. For example:
device = nb.dcim.devices.get(name='mydevice')
local_context = device.local_context_data
vlans_list = [{'vid': 1, 'name': 'vlan1'}, {'vid': 2, 'name': 'vlan2'}]
for vlan in vlans_list:
local_context['vlans'].append({"vid": vlan['vid'], 'name': vlan['name']})
device.update(dict(local_context_data=local_context))

device.update doesn't work, but if i don't use "get" from netbox, but set local_context in my code, it's work. For example:

device = nb.dcim.devices.get(name='pd100-elf-025')
new_local_context = {
"bgp_neighbors": [
{
"as": "65000",
"neighbor": "169.254.0.1"
},
{
"as": "65001",
"neighbor": "169.254.0.2"
}
],
"track": [
"10.10.0.251/32",
"10.10.0.252/32"
],
"vlans": []
}

vlans_list = [{'vid': 1, 'name': 'vlan1'}, {'vid': 2, 'name': 'vlan2'}]
for vlan in vlans_list:
new_local_context['vlans'].append({"vid": vlan['vid'], 'name': vlan['name']})
device.update(dict(local_context_data=new_local_context))

I tried it in postman:
If request metod is "put" - local_config_context will be replace
If request metod is "patch" - local_config_context will be update

**Comment by markkuleinio** [comment]
Your `update()` syntax looks strange. For me it works just fine like this:

```
>>> d = netbox.dcim.devices.get(...)
>>> d.local_context_data
{'vlans': []}
>>> local_context
{'vlans': [{'vid': 1, 'name': 'vlan1'}, {'vid': 2, 'name': 'vlan2'}]}
>>> d.update({"local_context_data": local_context})
True
>>> d.local_context_data
{'vlans': [{'vid': 1, 'name': 'vlan1'}, {'vid': 2, 'name': 'vlan2'}]}
>>>
```

**Comment by markkuleinio** [comment]
I don't know the exact reason, but I have a workaround. This line:

`local_context = device.local_context_data`

Replace it with:
```
import copy
local_context = copy.deepcopy(device.local_context_data)
```

and then you can modify `local_context` as you wish and `device.update()` works as expected.

Let us know how it goes for you.

**That said**, I don't exactly know why the diff in `save()` (behind the `update()` call) says that no changes were made to the object. Can @zachmoody tell why `Record._diff()` does not see the data change in this case? Based on my debug prints diff is False in `save()`, thus `req.patch()` is not called (verified from web server logs as well). 

Full test code:
```
import pynetbox
netbox = pynetbox.api("http://netbox-test.example.com/", token="xyz")
# Pick whatever device you need
device = netbox.dcim.devices.get(3)
# Ensure local context data is {"vlans": []} at start
print(device.local_context_data)
local_context = device.local_context_data
vlans_list = [{'vid': 1, 'name': 'vlan1'}, {'vid': 2, 'name': 'vlan2'}]
for vlan in vlans_list:
    local_context["vlans"].append({"vid": vlan['vid'], 'name': vlan['name']})
print(device.update(dict(local_context_data=local_context)))
# Returns False and no data change happens in NetBox
```

**Comment by nazarko-es** [comment]
Hello, thank you, i will trying to use your workaround

**Comment by nazarko-es** [comment]
Hello, workaround is working, thanks. If you will fix this bug, tell me please about it

**Comment by zachmoody** [comment]
It was fixed in #266.
